plugin/action/modify/modify.go

```diff
@@ -92,7 +100,12 @@ func (p *Plugin) Do(event *pipeline.Event) pipeline.ActionResult {
 			}
 		}
 
-		event.Root.AddFieldNoAlloc(event.Root, field).MutateToBytesCopy(event.Root, p.buf)
+		node := event.Root.Dig(fo.field...)
+		if node == nil {
+			node = event.Root.AddFieldNoAlloc(event.Root, field)
```

Comment:
Seems like it will add single field `field.subfield` but we want to add nested field `field`->`subfield`. And it would be great to have a test for that case.

Reply:
done